packages/core/src/components/fields/SchemaField.tsx

```diff
@@ -151,7 +151,7 @@ function SchemaFieldRender<T = any, S extends StrictRJSFSchema = RJSFSchema, F e
 
   const FieldComponent = getFieldComponent<T, S, F>(schema, uiOptions, idSchema, registry);
   const disabled = Boolean(uiOptions.disabled ?? props.disabled);
-  const readonly = Boolean(uiOptions.readonly ?? props.readonly ?? props.schema.readOnly ?? schema.readOnly);
+  const readonly = Boolean(uiOptions.readonly ?? (props.readonly || props.schema.readOnly || schema.readOnly));
```

Comment:
Fixes case where `readonly` should be true when `props.readonly` is `false` and `props.schema.readOnly` is `true`.

In plain terms, revert to the behavior before #4212, except allow `uiOptions.readOnly` to override other readonly values as originally intended